### PR TITLE
Benchmarks: Microbenchmark - Support different hipblasLt data types in dist_inference

### PR DESCRIPTION
**Pull request by yzygitzh**
**Description**
Support different data types in different hipblasLt versions for dist_inference

### CODE REVIEW

**Review by yukirora — APPROVED** [review]

File: superbench/benchmarks/micro_benchmarks/dist_inference_cpp/CMakeLists.txt [hunk 1]
```diff
@@ -31,6 +31,9 @@ else()
         # link hip device lib
         add_executable(dist_inference dist_inference.cpp)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2 -DROCM_USE_FLOAT16=1")
+        if(DEFINED ENV{HIPBLASLT_DATATYPE_WORKAROUND})
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DHIPBLASLT_DATATYPE_WORKAROUND=1")
+        endif()
         target_link_libraries(dist_inference MPI::MPI_CXX rccl hipblaslt hip::device)
     else()
         message(FATAL_ERROR "No CUDA or ROCm environment found.")
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/dist_inference_cpp/CMakeLists.txt:39]: Please merge it in bugbash branch.

Comment (yzygitzh) [reply superbench/benchmarks/micro_benchmarks/dist_inference_cpp/CMakeLists.txt:39]: Fixed

File: superbench/benchmarks/micro_benchmarks/dist_inference_cpp/dist_inference.cu [hunk 2]
```diff
@@ -45,6 +45,13 @@
 #include <hipblaslt/hipblaslt.h>
 #include <rccl/rccl.h>
 using cublasLtHalf = hipblasLtHalf;
+#if defined(HIPBLASLT_DATATYPE_WORKAROUND)
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/dist_inference_cpp/dist_inference.cu:48]: @yukirora please check whether we should find a version to switch it.

Comment (yzygitzh) [reply superbench/benchmarks/micro_benchmarks/dist_inference_cpp/dist_inference.cu]: Fixed naming.

**Pull request merged**